### ISSUE TITLE
Skia crash in customer:money

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#69642]
Internal: b/172177383

customer:money reported crash seen in the wild.

```
Thread 0  (id: 0x00000303) CRASHED [EXC_BAD_ACCESS / KERN_INVALID_ADDRESS @ 0x000002e0 ]Show exception record
0x0000000182e17a74(Metal + 0x00010a74)
0x00000001a0963190(AGXMetalA8 + 0x0007f190)
0x00000001a095c3f4(AGXMetalA8 + 0x000783f4)
0x00000001a093ea58(AGXMetalA8 + 0x0005aa58)
0x0000000105340708(Flutter + 0x002b4708)
0x0000000105347e48(Flutter + 0x002bbe48)
0x0000000105348458(Flutter + 0x002bc458)
0x0000000105340fa4(Flutter + 0x002b4fa4)
0x000000010525944c(Flutter + 0x001cd44c)
0x000000010524cad8(Flutter + 0x001c0ad8)
0x000000010524cdf8(Flutter + 0x001c0df8)
0x00000001052787c8(Flutter + 0x001ec7c8)
0x00000001052ef5a4(Flutter + 0x002635a4)
0x000000010550b5cc(Flutter + 0x0047f5cc)
0x000000010511b60c(Flutter + 0x0008f60c)
0x00000001050a5268(Flutter + 0x00019268)
0x00000001050bee8c(Flutter + 0x00032e8c)
0x000000010539e840(Flutter + 0x00312840)
0x000000010539f400(Flutter + 0x00313400)
0x000000010539eb10(Flutter + 0x00312b10)
0x00000001053a5208(Flutter + 0x00319208)
0x00000001050c9c7c(Flutter + 0x0003dc7c)
0x00000001050cbbf0(Flutter + 0x0003fbf0)
0x00000001810e3dbc(CoreFoundation + 0x000ebdbc)
0x00000001810e3ae0(CoreFoundation + 0x000ebae0)
0x00000001810e32e0(CoreFoundation + 0x000eb2e0)
0x00000001810e0ec8(CoreFoundation + 0x000e8ec8)
0x0000000181000c54(CoreFoundation + 0x00008c54)
0x0000000182eacf80(GraphicsServices + 0x0000af80)
0x000000018a7595c0(UIKit + 0x000735c0)
0x00000001003cb6b4(ios_prod_global -main.m:35)main
0x0000000180b20568(libdyld.dylib + 0x00001568)
```

Unfortunately, user reports are all we have at this point. No reproduction. We do know that this particular crash came from iOS 11.2 device. They have other traces as well which end up in AGXMetalA8 lib and together they form ~7% of their total crashes on iOS (2nd most severe) making this a P2.

**Comment by mehmetf** [comment]
@flutter-symbolizer-bot 2eac514f26a676d1df0821e815d501149d3ead7f release arm64

**Comment by mehmetf** [comment]
@mraleph How does the bot know what platform to use? From the crash format?

**Comment by mraleph** [comment]
@mehmetf yeah, from the crash format. also you need to be a public member of Flutter org on GitHub (which you are not - you can check by looking at this issue in incognito window) to issue commands to the bot. 

I will do some tweaks to the bot and update this issue once it is ready.

**Comment by mraleph** [comment]
I have made some tweaks to the bot, including writing a README for it (check its profile). 

What follows is the demo.

**Comment by mraleph** [comment]
@flutter-symbolizer-bot 2eac514f26a676d1df0821e815d501149d3ead7f release arm64

**Comment by mraleph** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/69642#issue-735022434 internal engine#2eac514f26a676d1df0821e815d501149d3ead7f ios release arm64

**Comment by mraleph** [comment]
@mehmetf as you can see I added support for internal stack trace format for ios. If you have examples of how internal stacks for Android apps look like, please send them to me as well so that I could adjust RegExps. Bot now reports errors to users that try to invoke it.

**Comment by mraleph** [comment]
@xster seems like a Skia crash on iOS reporter by @mehmetf see symbolization above.

**Comment by mehmetf** [comment]
@chinmaygarde as well. Unfortunately, user reports are all we have at this point. No reproduction. We do know that this particular crash came from iOS 11.2 device.  They have other traces as well which end up in AGXMetalA8 lib and together they form ~7% of their total crashes on iOS (2nd most severe) making this a P2.

**Comment by mehmetf** [comment]
This is the other trace:

```
Thread 0  (id: 0x00000303) CRASHED [EXC_BAD_ACCESS / KERN_INVALID_ADDRESS @ 0x000002e0 ]Show exception record
0x000000018733fa74(Metal + 0x00010a74)
0x00000001a4d808c0(AGXMetalA8 + 0x000728c0)
0x00000001a4d689bc(AGXMetalA8 + 0x0005a9bc)
0x0000000105b70534(Flutter + 0x002b4534)
0x0000000105b73610(Flutter + 0x002b7610)
0x0000000105a81f24(Flutter + 0x001c5f24)
0x0000000105a885d8(Flutter + 0x001cc5d8)
0x0000000105a7a630(Flutter + 0x001be630)
0x0000000105a7a5b4(Flutter + 0x001be5b4)
0x0000000105a88270(Flutter + 0x001cc270)
0x0000000105a88200(Flutter + 0x001cc200)
0x0000000105a7c9c4(Flutter + 0x001c09c4)
0x0000000105a7cdf8(Flutter + 0x001c0df8)
0x0000000105aa87c8(Flutter + 0x001ec7c8)
0x0000000105b1f5a4(Flutter + 0x002635a4)
0x0000000105d3b5cc(Flutter + 0x0047f5cc)
0x000000010594b60c(Flutter + 0x0008f60c)
0x00000001058d5268(Flutter + 0x00019268)
0x00000001058eee8c(Flutter + 0x00032e8c)
0x0000000105bce840(Flutter + 0x00312840)
0x0000000105bcf400(Flutter + 0x00313400)
0x0000000105bceb10(Flutter + 0x00312b10)
0x0000000105bd5208(Flutter + 0x00319208)
0x00000001058f9c7c(Flutter + 0x0003dc7c)
0x00000001058fbbf0(Flutter + 0x0003fbf0)
0x000000018560afbc(CoreFoundation + 0x000eefbc)
0x000000018560ace0(CoreFoundation + 0x000eece0)
0x000000018560a4e0(CoreFoundation + 0x000ee4e0)
0x00000001856080c8(CoreFoundation + 0x000ec0c8)
0x0000000185527e54(CoreFoundation + 0x0000be54)
0x00000001873d4f80(GraphicsServices + 0x0000af80)
0x000000018eba7678(UIKit + 0x00073678)
0x0000000100d8f6b4(ios_prod_global -main.m:35)main
0x0000000185044568(libdyld.dylib + 0x00001568)
```

**Comment by mehmetf** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/69642#issuecomment-721246638 internal engine#2eac514f26a676d1df0821e815d501149d3ead7f ios release arm64

**Comment by xster** [comment]
Has signs of https://github.com/flutter/flutter/issues/69449. @mehmetf does money use platform views?

**Comment by chinmaygarde** [comment]
The second stack trace looks like a duplicate of https://github.com/flutter/flutter/issues/69102. This is unlikely to be the platform-views race that @xster linked to.

**Comment by xster** [comment]
Ah never mind you're right. @gaaclarke already looked at it and seems like it's another background GPU issue?

**Comment by gaaclarke** [comment]
Yea, I suspect this may be a background GPU problem.  Looks like a canvas flush is causing some GPU operations and it isn't protected with the GPU-disabling SyncSwitch.  I can create a PR today.

https://github.com/flutter/flutter/issues/69102 is a bit different because the syncswitch was in the stacktrace.

**Comment by gaaclarke** [comment]
This is probably another issue that was caused by https://github.com/flutter/engine/pull/20671  We are now sending people through the platform views controller when we didn't previously didn't and that code was meant to run in a situation where the main thread and the raster thread were merged so we couldn't get GPU operations in the background.  I suspect @cyanglaz's PR https://github.com/flutter/engine/pull/22275 will fix this.

**Comment by gaaclarke** [comment]
Actually, if you look at the thread, it is thread 0, which the platform thread.  This means this crash is happening when the threads are merged.  If this is a GPU backgrounding problem, that means we are getting some events on the platform thread after `applicationWillResignActive`.  So, the GPU is disabled, but since this doesn't go through the syncswitch it still tries to use the GPU.  I was wrong, we need to route this code through the gpu disable syncswitch despite the threads being merged.

**Comment by mehmetf** [comment]
Even with the fixes, customer:money continues to see this crash.

```
Thread 7  (id: 0x00007103) CRASHED [EXC_BAD_ACCESS / KERN_INVALID_ADDRESS @ 0x000002e0 ]Show exception record
0x0000000186d207fc(Metal + 0x0000f7fc)
0x00000001a557a85c(AGXMetalA7 + 0x0007985c)
0x00000001a550b614(AGXMetalA7 + 0x0000a614)
0x00000001a550d5c8(AGXMetalA7 + 0x0000c5c8)
0x00000001063d0724(Flutter + 0x002b8724)
0x00000001063d36ec(Flutter + 0x002bb6ec)
0x00000001062e3160(Flutter + 0x001cb160)
0x00000001062e2ddc(Flutter + 0x001caddc)
0x00000001062fa2a0(Flutter + 0x001e22a0)
0x00000001062ee184(Flutter + 0x001d6184)
0x0000000106305994(Flutter + 0x001ed994)
0x00000001062edd50(Flutter + 0x001d5d50)
0x00000001062c85a8(Flutter + 0x001b05a8)
0x0000000106384bfc(Flutter + 0x0026cbfc)
0x00000001064d07cc(Flutter + 0x003b87cc)
0x00000001061562e0(Flutter + 0x0003e2e0)
0x00000001064d00ac(Flutter + 0x003b80ac)
0x0000000106154fc4(Flutter + 0x0003cfc4)
0x0000000106156f38(Flutter + 0x0003ef38)
0x0000000184eb3aa4(CoreFoundation + 0x000eeaa4)
0x0000000184eb3768(CoreFoundation + 0x000ee768)
0x0000000184eb300c(CoreFoundation + 0x000ee00c)
0x0000000184eb0b5c(CoreFoundation + 0x000ebb5c)
0x0000000184dd0da4(CoreFoundation + 0x0000bda4)
0x0000000106156e14(Flutter + 0x0003ee14)
0x000000010615679c(Flutter + 0x0003e79c)
0x0000000184b3121c(libsystem_pthread.dylib + 0x0000221c)
0x0000000184b3110c(libsystem_pthread.dylib + 0x0000210c)
```

**Comment by mehmetf** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/69642#issuecomment-738887789 internal engine#5dcf1d243eb401d59dab2b6310dea36267c2ba7c ios release arm64

**Comment by mehmetf** [comment]
This engine hash was generated by cherrypicking this fix (and a couple of others) onto the dev branch: https://github.com/flutter/engine/commits/5dcf1d243eb401d59dab2b6310dea36267c2ba7c

**Comment by mybotgb** [comment]
Hi , Can you provide the EAT for this fix to be released on stable channel? It really causes many crash on ios

**Comment by Plegizplegiz** [comment]
ma 7.12.2020 klo 5.45 mybotgb <notifications@github.com> kirjoitti:

**Comment by xster** [comment]
@gaaclarke could you give this one another look?

**Comment by gaaclarke** [comment]
The SyncSwitch is in the stacktrace so this isn't a result of executing GPU commands in the background like we thought.  This now is a duplicate of https://github.com/flutter/flutter/issues/69102 which chinmay is looking into.  On that issue I asked if anyone had the console log (https://github.com/flutter/flutter/issues/69102#issuecomment-720738001), maybe someone will have them here.

**Comment by gaaclarke** [comment]
I'm going to close this as a dup of #69102 and provide any further updates there.
